pkg/util/cluster/cluster.go

```diff
@@ -482,11 +483,15 @@ func (c *Cluster) deleteRoleAssignments(ctx context.Context, appID string) error
 		return err
 	}
 
+	scopeResourceGroupOrLower := regexp.MustCompile("^/subscriptions/.*?/resourcegroups/")
 	for _, roleAssignment := range roleAssignments {
-		c.log.Infof("deleting role assignment %s", *roleAssignment.Name)
-		_, err = c.roleassignments.Delete(ctx, *roleAssignment.Scope, *roleAssignment.Name)
-		if err != nil {
-			return err
+		if scopeResourceGroupOrLower.MatchString(strings.ToLower(*roleAssignment.Scope)) {
```

Comment:
simplifies to
```golang
// skip subscription-level scopes (/subscriptions/xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx)
if strings.Count(*roleAssignment.Scope, "/") < 3 {
    continue
}
```

1) generally, avoid regexps when you can
2) if foo { continue } is often preferable to excessive nesting

Reply:
There are some role assignments which have >= 3 slashes which we might not want to target, such as `/providers/Microsoft.Management/managementGroups/<UUID>`. I don't know if it would try to or be able to remove those, though.

Reply:
switched to `strings.HasPrefix` to simplify